app/models/uphold_connection.rb

```diff
@@ -202,11 +202,24 @@ def valid_card?
     false
   end
 
+  def refresh_token
+    JSON.parse(uphold_access_parameters || '{}').fetch('refresh_token', nil)
+  end
+
+  def authorization_expires_at
+    JSON.parse(uphold_access_parameters || '{}').fetch('expiration_time', nil)&.to_datetime
+  end
+
+  def authorization_expired?
+    authorization_expires_at&. > Time.zone.now
```

Comment:
```suggestion
    authorization_expires_at.present? && authorization_expires_at > Time.zone.now
```

This should always guarantee a boolean return type